### PR TITLE
Add DialHTTPPath with custom dialer to lib/net.

### PR DESCRIPTION
**Pull request by keep94**

### CODE REVIEW

**Comment by rgooch** [comment]
I would rather see this go into a new lib/net/rpc package, so that lib/net doesn't bring in the net/rpc standard library package.

**Comment by keep94** [comment]
Ok I will rework it

File: lib/net/rpc.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"errors"
+	"io"
+	gonet "net"
+	"net/http"
+	"net/rpc"
+)
+
+func dialHTTPPath(dialer Dialer, network, address, path string) (
+	*rpc.Client, error) {
+	var err error
```
Comment (rgooch) [inline lib/net/rpc.go]: You may want to consider letting the caller pass an empty string and default to `rpc.DefaultRPCPath`

Comment (keep94) [inline lib/net/rpc.go]: I thought about that, but figured there is value in letting this DialHTTPPath work just like the one in golang which does not provide for passing an empty string for path.  Besides, it looks like you already change an empty path to rpc.DefaultRPCPath in rpcclientpool package. So it doesn't seem right to have that logic twice.

**Comment by keep94** [comment]
Ok, I moved this code into lib/net/rpc. Let me know if you still want me to allow user to pass empty string for path.

File: lib/net/rpc.go (hunk 1 above)
Comment (rgooch) [inline lib/net/rpc.go]: OK, fair enough.

**Comment by keep94** [comment]
Ok. I triggered another build.

**Pull request merged**